app/models/manageiq/providers/openshift/container_manager/container_image.rb

```diff
@@ -1,10 +1,44 @@
 class ManageIQ::Providers::Openshift::ContainerManager::ContainerImage < ContainerImage
-  def annotate_deny_execution(causing_policy)
+  def annotate_image(annotations)
     ext_management_system.annotate(
       "image",
       digest,
-      "security.manageiq.org/failed-policy" => causing_policy,
-      "images.openshift.io/deny-execution"  => "true"
+      annotations
     )
   end
+
+  def openscap_summary
+    failed_rules = openscap_rule_results.where(:result => "fail").group(:severity).count
+    [[['High'], 'Critical', 3],
+     [['Medium'], 'Important', 2],
+     [['Low'], 'Medium', 1],
+     [['Info', 'Unknown'], 'Low', 0]].collect do |severities, label, index|
+      {
+        :label         => label,
+        :severityIndex => index,
+        :data          => failed_rules.select { |sev| severities.include?(sev) }.values.sum
+      }
+    end
+  end
+
+  def security_quality_annotation(compliant)
+    {"quality.images.openshift.io/vulnerability.openscap" => {
+      :name        => "ManageIQ",
+      :timestamp   => Time.now.utc.to_i,
+      :description => "OpenSCAP Score",
+      :reference   => Rails.application.routes.url_helpers.url_for(:controller => "container_image",
+                                                                   :action     => "openscap_html",
+                                                                   :id         => id,
+                                                                   :only_path  => true),
```

Comment:
The annotation will live in Openshift, so a relative URL isn't very useful.
Would `:only_path => false` give an absolute URL?